io_gripper_controller/src/io_gripper_controller.cpp

```diff
@@ -0,0 +1,1369 @@
+                                        std::unordered_set<std::string> & interface_list)
+  {
+    for (const auto & itf : parameter_values)
+    {
+      if (!itf.empty())
+      {
+        ios[itf] = value;
+        interface_list.insert(itf);
+      }
+    }
+  };
+
+  // make full command ios lists -- just once
+  parse_interfaces_from_params(
+    params_.open.set_before_command.high, 1.0, open_ios_.set_before_command_ios, command_if_ios);
+  parse_interfaces_from_params(
+    params_.open.set_before_command.low, 0.0, open_ios_.set_before_command_ios, command_if_ios);
+
+  parse_interfaces_from_params(
+    params_.open.command.high, 1.0, open_ios_.command_ios, command_if_ios);
+  parse_interfaces_from_params(
+    params_.open.command.low, 0.0, open_ios_.command_ios, command_if_ios);
+
+  parse_interfaces_from_params(
+    params_.open.set_after_command.high, 1.0, open_ios_.set_after_command_ios, command_if_ios);
+  parse_interfaces_from_params(
+    params_.open.set_after_command.low, 0.0, open_ios_.set_after_command_ios, command_if_ios);
+
+  for (const auto & key : params_.close.set_before_command.high)
+  {
+    if (!key.empty())
+    {
+      set_before_command_close.push_back(key);
+      set_before_command_close_values.push_back(1.0);
+      command_if_ios.insert(key);
+    }
+  }
+
+  for (const auto & key : params_.close.set_before_command.low)
+  {
+    if (!key.empty())
+    {
+      set_before_command_close.push_back(key);
+      set_before_command_close_values.push_back(0.0);
+      command_if_ios.insert(key);
+    }
+  }
+
+  for (const auto & key : params_.close.command.high)
+  {
+    command_ios_close.push_back(key);
+    command_ios_close_values.push_back(1.0);
+    command_if_ios.insert(key);
+  }
+
+  for (const auto & key : params_.close.command.low)
+  {
+    if (!key.empty())
+    {
+      command_ios_close.push_back(key);
+      command_ios_close_values.push_back(0.0);
+      command_if_ios.insert(key);
+    }
+  }
+
+  // make full state ios lists -- just once
+  parse_interfaces_from_params(params_.open.state.high, 1.0, open_ios_.state_ios, state_if_ios);
+  parse_interfaces_from_params(params_.open.state.low, 0.0, open_ios_.state_ios, state_if_ios);
+
+  for (const auto & [name, value] : params_.close.state.possible_closed_states_map)
+  {
+    for (const auto & key : value.high)
+    {
+      if (!key.empty())
+      {
+        state_if_ios.insert(key);
+      }
+    }
+    for (const auto & key : value.low)
+    {
+      if (!key.empty())
+      {
+        state_if_ios.insert(key);
+      }
+    }
+    for (const auto & key : value.set_after_command_high)
+    {
+      if (!key.empty())
+      {
+        command_if_ios.insert(key);
+      }
+    }
+    for (const auto & key : value.set_after_command_low)
+    {
+      if (!key.empty())
+      {
+        command_if_ios.insert(key);
+      }
+    }
+  }
+
+  // get the configurations for different io which needs to be high or low
```

Comment:
For the configuration, we should also use the `GripperTransitionIOs` structure. For now, convert only these variables here, but subsequently, we will also use `command before` and `commands after`.